### ISSUE TITLE
Accuracy of sample_uff_maskrcnn decreases as Pre_NMS_LImit is set different

### ISSUE THREAD

**Issue by ManishSoni1908** [NVIDIA/TensorRT#1263]
## Description
I have converted the maskrcnn keras model to .uff using the code in the sampleUFFMaskRcnn folder. But results are different and accuracy is low as **prenms_topk** is set to 1024 in the proposal layer in the config file.

**prenms_topk** named pre_nms_limit in matterport keras repo st to 6000, I found we cannot set 6000 as in kernel is written maximum to use 4096, I tried with setting up prenms_topk to 4096 and 2048 but it gives the following error.

[04/22/2021-23:40:03] [E] [TRT] engine.cpp (859) - Cuda Error in executeInternal: 77 (an illegal memory access was encountered)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeContext.cpp (133) - Cudnn Error in configure: 7 (CUDNN_STATUS_MAPPING_ERROR)
[04/22/2021-23:40:03] [E] [TRT] FAILED_EXECUTION: std::exception
[04/22/2021-23:40:03] [I] Run for 10 times with Batch Size 1
[04/22/2021-23:40:03] [I] Average inference time is 7.86446 ms/frame
[04/22/2021-23:40:03] [E] [TRT] engine.cpp (165) - Cuda Error in ~ExecutionContext: 77 (an illegal memory access was encountered)
[04/22/2021-23:40:03] [E] [TRT] INTERNAL_ERROR: std::exception
[04/22/2021-23:40:03] [E] [TRT] Parameter check failed at: …/rtSafe/safeContext.cpp::terminateCommonContext::165, condition: cudaEventDestroy(context.start) failure.
[04/22/2021-23:40:03] [E] [TRT] Parameter check failed at: …/rtSafe/safeContext.cpp::terminateCommonContext::170, condition: cudaEventDestroy(context.stop) failure.
[04/22/2021-23:40:03] [E] [TRT] …/rtSafe/safeRuntime.cpp (32) - Cuda Error in free: 77 (an illegal memory access was encountered)
terminate called after throwing an instance of ‘nvinfer1::CudaError’
what(): std::exception
Aborted (core dumped)

## Environment

**TensorRT Version**:  6.0.1
**NVIDIA GPU**: 1080TI
**NVIDIA Driver Version**: 418 
**CUDA Version**: 10.0
**CUDNN Version**: 7.6
**Operating System**: Ubutnu
**Python Version (if applicable)**: 3.5 
**Tensorflow Version (if applicable)**: 1.13
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by ttyio** [comment]
@Tyler-D  ^^

**Comment by Tyler-D** [comment]
I think you can compile `Debug` version of plugin lib to activate the `assert` function in the kernel. Then you will see if is related to the maskrcnn kernels.

And if you just want to use MaskRCNN with TensorRT, I suggest you to try [Transfer Learning Toolkit](https://docs.nvidia.com/metropolis/TLT/tlt-user-guide/text/instance_segmentation/mask_rcnn.html).

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks. please reopen if you still have question, thanks!
